groupcache.go

```diff
@@ -320,11 +328,33 @@ func (g *Group) load(ctx context.Context, key string, dest Sink) (value ByteView
 		var value ByteView
 		var err error
 		if peer, ok := g.peers.PickPeer(key); ok {
+
+			// metrics duration start
+			start := time.Now()
+
+			// get value from peers
 			value, err = g.getFromPeer(ctx, peer, key)
+
+			// metrics duration compute
+			duration := time.Since(start).Milliseconds()
+
+			// metrics only store the slowest duration
+			if g.Stats.GetFromPeersSlowestDuration.Get() < duration {
+				g.Stats.GetFromPeersSlowestDuration.Store(duration)
+			}
+
 			if err == nil {
 				g.Stats.PeerLoads.Add(1)
 				return value, nil
 			}
+
+			if logger != nil {
+				logger.WithFields(logrus.Fields{
+					"err": err,
+					"key": key,
+				}).Error("groupcache: error retrieving key from peers")
```

Comment:
Would be nice to include the peer such that log message looks like
```
error retrieving key from peer 'scout-n01'
```
Also, instead of prefix message with `groupcache:` we add a category field. `"category": "groupcache"` which makes it easy to search ELK stack for groupcache related errors.

Reply:
Regarding : 

```
Would be nice to include the peer such that log message looks like
  > error retrieving key from peer 'scout-n01'
```

Do you want :
* the peer's hostname that make the call to the remote peers

Or : 
* the remote peer’s hostname that generate the error when being called

So,
If it is `the peer's hostname that make the call to the remote peers` 
when setting the logger with `SetLogger()`, the logger can be setup with a default field like this :

```
logger = logger.WithFields(logrus.Fields{
	"peer": hostname,
})
groupcache.SetLogger(logger)
```

if it is `the remote peer’s hostname that generate the error when being called`
I don't see how to do it without modifying the code a lot ?

Reply:
ELK stacks should take care of identifying where the log entry is coming from. `error retrieving key from peer 'scout-n01'` is sufficient.

Reply:
The `peer` variable should have the hostname/ip of the remote peer.

Reply:
peer is a variable of type ProtoGetter which is an interface 

```go
type ProtoGetter interface {
	Get(context context.Context, in *pb.GetRequest, out *pb.GetResponse) error
	Remove(context context.Context, in *pb.GetRequest) error
}
```

So we don't have access to any private field that might contains the return var, Am I wrong, or miss something ?

Reply:
The implementation is `httpGetter`, you can add a new method `GetURL()` to the `ProtoGetter` interface which will return the peer url.  This interface is only used internal to groupcache for contacting peers. Does that make sense?

Reply:
ok gotcha,

error now will looks like : 
```
ERRO[0035] error retrieving key from peer 'http://MBP-tommy-42.local:5071'  category=groupcache err="dial tcp 127.0.0.1:5071: connect: connection refused" key="blabla" peer=MBP-tommy-42.local
```